main.go

```diff
@@ -462,3 +481,294 @@ func stopStalePipelines (client *gitlab.Client, vars []*gitlab.PipelineVariable)
+				"and creating the PR, so that the user can manually resolve, " +
+				"and re-push to the PR once these are fixed"
+			log.Warn(msg)
+		} else {
+			return false, fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+		}
+	}
+
+	if !merged {
+		// In case of a failed merge, reset PRBranchName to opensource/branchName
+		// and push this branch to enterprise
+		gitcmd = exec.Command("git", "reset", "--hard", "opensource/"+branchName)
+		gitcmd.Dir = tmpdir
+		out, err = gitcmd.CombinedOutput()
+		if err != nil {
+			return merged, fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+		}
+	}
+
+	// Push the branch to the mender-test-bot's own fork
+	gitcmd = exec.Command("git", "push", "--set-upstream", "mender-test-bot", PRBranchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return merged, fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	if merged {
+		log.Infof("Merged branch: opensource/%s/%s into enterprise/%[1]s/%s in the Enterprise repo",
+			repo, branchName, PRBranchName)
+	} else {
+		msg := "Failed to merge opensource/%s/%s into enterprise/%[1]s/%s in the Enterprise repo. " +
+			"Therefore pushed opensource/%[1]s/%[2]s to %s so that " +
+			"merging can be done by a human locally"
+		log.Infof(msg, repo, branchName, PRBranchName, PRBranchName)
+	}
+
+	return merged, nil
+}
+
+type createPRArgs struct {
+	conf        *config
+	repo        string
+	prBranch    string
+	baseBranch  string
+	message     string
+	messageBody string
+}
+
+func createPullRequestFromTestBotFork(args createPRArgs) (*github.PullRequest, error) {
+
+	client := createGitHubClient(args.conf)
+
+	newPR := &github.NewPullRequest{
+		Title:               github.String(args.message),
+		Head:                github.String(args.prBranch),
+		Base:                github.String(args.baseBranch),
+		Body:                github.String(args.messageBody),
+		MaintainerCanModify: github.Bool(true),
+	}
+
+	pr, _, err := client.PullRequests.Create(context.Background(), "mendersoftware", args.repo, newPR)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create the PR for: (%s) %v", args.repo, err)
+	}
+
+	return pr, nil
+}
+
+type commentArgs struct {
+	pr             *github.PullRequest
+	conf           *config
+	mergeConflicts bool
+	repo           string
+	userName       string
+}
+
+func commentToNotifyUser(args commentArgs) error {
+
+	// Post a comment, and @mention the user
+	var commentBody string
+	if !args.mergeConflicts {
+		commentBody = fmt.Sprintf("@%s I have created a PR for you, ready to merge as soon as tests are passed", args.userName)
+	} else {
+		msg := "@%s I have created a PR for you. Unfortunately there were some merge conflicts " +
+			"which I failed to resolve automatically. Therefore I have created this PR for you, " +
+			"which is ready to be merged locally, and then pushed here :)"
```

Comment:
Nice and friendly message :)

We are saying twice "I have created a PR for you", though. Maybe the second time we can just say "Therefore the branch of this PR is ready to be merged locally, and then pushed here".

I don't really know... the process is going to be complex anyway (because you need the two remotes trick).

Reply:
> Nice and friendly message :)
> 
> We are saying twice "I have created a PR for you", though. Maybe the second time we can just say "Therefore the branch of this PR is ready to be merged locally, and then pushed here".

Yes, two times is a little much. I like your suggestion. Fixed!
> 
> I don't really know... the process is going to be complex anyway (because you need the two remotes trick).

I don't follow on this one?

Reply:
> I don't follow on this one. What is it you don't know?

I will figure it out the first time I need to resolve conflicts :P

Reply:
haha, alright! Let's hope it's not as unintuitive as you fear.